envoy/conf.yaml.example

```diff
@@ -0,0 +1,33 @@
+# This file is overwritten upon Agent upgrade.
+# To make modifications to the check configuration, please copy this file
+# to `envoy.yaml` and make your changes on that file.
+
+init_config:
+
+instances:
+  # For every instance, you need a `stats_url` and can optionally
+  # supply a list of tags. The admin endpoint must be accessible.
+  # https://www.envoyproxy.io/docs/envoy/latest/operations/admin
+
+  - stats_url: http://localhost:80/stats
```

Comment:
Hi @htuch and @taiki45,

Thanks for sharing your concerns with us! We understand them, and this is what we are doing at the moment to address them:

1. Until envoyproxy/envoy#2763 is resolved, we are offering our customers an [alternative way](https://github.com/DataDog/integrations-core/tree/ofek/envoy/envoy#secured-stats-endpoint) to access `Envoy`'s `/stats` endpoint.
2. Our `envoy` integration supports SSL verification of the `Envoy` endpoint (and this requires `Envoy` to support SSL). If the SSL verification fails for any reason when enabled, the integration will not collect metrics.
3. Our integration has the option to support [basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) (but this requires `Envoy` to implement basic authentication).
4. Our integration does not need to be aware of RBAC, but when `Envoy` implements it, our integration will use it transparently (via basic authentication).

At the moment, we do not see the benefit of supporting the `dogstatsd` connector, because that still does not prevent the admin endpoint on `Envoy` from being exposed to the entire trusted network. A customer could use that to be more secure in some network scenarios, but we would also like to provide our customers with a working agent integration for `Envoy` in its current form. 

We also need to do some massaging of metrics in the integration for our backend, which the `dogstatsd` connector currently does not allow. Additionally, we are finding our parser a bit more resilient to stat name collisions. @jmarantz mentioned that he has a similar implementation for which he’ll issue a PR to `Envoy` soon.

We would be happy to enforce authentication by default, provided that `Envoy` implements it. We hope this addresses your concerns. Thanks again for sharing your thoughts with us!

Regards,
The Agent team at Datadog

Reply:
Yeah, this all seems very reasonable. I'd add a couple of points for consideration:
* With your example at https://github.com/DataDog/integrations-core/tree/ofek/envoy/envoy#secured-stats-endpoint, should you have the admin bind only to 127.0.0.1 to avoid incoming connections from NICs?
* I probably worded the above a little strongly; until https://github.com/envoyproxy/envoy/issues/2763 is resolved we don't have a great story for access outside a trusted network (although I like your example as a temporary workaround, you could post it to https://github.com/envoyproxy/envoy/issues/2763 once the above fix is made). We've recently been discussing this in  envoyproxy/data-plane-api#523 for Hystrix and agreed to separate out the admin security concerns from pull-based stats.

Reply:
@htuch That's an interesting idea. Would I simply change `"address": "tcp://0.0.0.0:8001"` to `"address": "tcp://127.0.0.1:8001"`?

Reply:
@ofek Yes, that's right. And I assume that prevents the entire admin endpoints from being exposed.

I agree to move to pull-based stats because it can be easily scaled than push-based one, but I'd leave some comments to clarify the points.

>At the moment, we do not see the benefit of supporting the dogstatsd connector, because that still does not prevent the admin endpoint on Envoy from being exposed to the entire trusted network. A customer could use that to be more secure in some network scenarios, but we would also like to provide our customers with a working agent integration for Envoy in its current form.

AFAIK, the datadog agent will be running in the customer's host (the same host as running Envoy process). If it's correct, we might be able to send stats to datadog agent without opening any Envoy admin endpoints. The scenario is:

- Configure Envoy admin endpoint to bind local loopback address (127.0.0.1 in IPv4).
- Send Envoy stats to datadog agent running on the same host using UDP with dog_statsd sink.

>We also need to do some massaging of metrics in the integration for our backend, which the dogstatsd connector currently does not allow.

I suppose you can do that modifications with writing a statsd relay component which receives stats and modifies the stats then re-pushes to somewhere (probably datadog agent?).

IIUC, the current admin's `/stats` endpoint is missing histogram metrics. https://github.com/envoyproxy/envoy/issues/1947 This can be resolved with https://github.com/envoyproxy/envoy/pull/2736 which will implement pull-based stats with stats sink (I didn't fully take a look the PR yet).

Reply:
@ofek ack.